flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannelTest.java

```diff
@@ -158,27 +160,31 @@ public void testConcurrentConsumeMultiplePartitions() throws Exception {
 		// Test
 		try {
 			// Submit producer tasks
-			List<Future<?>> results = Lists.newArrayListWithCapacity(
+			List<CompletableFuture<?>> results = Lists.newArrayListWithCapacity(
 				parallelism + 1);
 
 			for (int i = 0; i < parallelism; i++) {
-				results.add(executor.submit(partitionProducers[i]));
+				results.add(CompletableFuture.supplyAsync(
+					CheckedSupplier.unchecked(partitionProducers[i]::call), executor));
 			}
 
 			// Submit consumer
 			for (int i = 0; i < parallelism; i++) {
-				results.add(executor.submit(
-					new TestLocalInputChannelConsumer(
-						i,
-						parallelism,
-						numberOfBuffersPerChannel,
-						networkBuffers.createBufferPool(parallelism, parallelism),
-						partitionManager,
-						new TaskEventDispatcher(),
-						partitionIds)));
+				results.add(CompletableFuture.supplyAsync(
+					CheckedSupplier.unchecked(
+						new TestLocalInputChannelConsumer(
```

Comment:
Please introduce a dedicated variable for the consumer to make this code more readable, i.e.
```
final TestLocalInputChannelConsumer consumer = ....
results.add(CompletableFuture. .... (consumer::call)...);
```

Reply:
done